azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java

```diff
@@ -206,12 +198,31 @@ public com.azure.resourcemanager.appcontainers.models.ContainerApp updateResourc
             .map(action -> action.bind(this))
             .orElse(null);
         messager.success(AzureString.format("Container App({0}) is successfully updated.", getName()), browse);
+        printSuccessMessages();
         if (isImageModified) {
             AzureTaskManager.getInstance().runOnPooledThread(() -> this.getRevisionModule().refresh());
         }
         return result;
     }
 
+    private void printSuccessMessages() {
+        final Action<String> learnMore = Optional.ofNullable(AzureActionManager.getInstance().getAction(Action.OPEN_URL).withLabel("Learn More"))
+            .map(action -> action.bind("https://aka.ms/azuretools-aca-stack"))
+            .orElse(null);
+        final Action<String> openPortal = Optional.ofNullable(AzureActionManager.getInstance().getAction(Action.OPEN_URL).withLabel("Open Portal"))
+            .map(action -> action.bind(this.getPortalUrl()))
+            .orElse(null);
+        final Action<String> openApp = Optional.ofNullable(this.getIngressFqdn())
+            .filter(fqdn -> !StringUtils.isEmpty(fqdn))
+            .flatMap(fqdn -> Optional.ofNullable(AzureActionManager.getInstance()
+                    .getAction(Action.OPEN_URL)
+                    .withLabel("Open Application"))
+                .map(action -> action.bind(String.format("https://%s", fqdn))))
+            .orElse(null);
+
+        AzureMessager.getMessager().info("To take advantage of this Java-optimized feature, please set your development stack to `Java` in the portal.", learnMore, openPortal, openApp);
```

Comment:
Shall we remove this message? Given that in the next stage we are going to set this runtime property for Java apps.

Reply:
Sure. Will remove this after java stack is enabled automatically.

Reply:
`this` --> `the`

It's not obvious what `this` points to.